pandas/src/parse_helper.h

```diff
@@ -39,22 +42,35 @@ int floatify(PyObject* str, double *result, int *maybe_int) {
 
     if (!status) {
         /* handle inf/-inf */
-        if (0 == strcmp(data, "-inf")) {
-            *result = -HUGE_VAL;
-            *maybe_int = 0;
-        } else if (0 == strcmp(data, "inf")) {
-            *result = HUGE_VAL;
-            *maybe_int = 0;
+        if (strlen(data) == 3 || strlen(data) == 4) {
+            strcpy(inf_str, data);
+            for ( ; *inf_ptr; ++inf_ptr) *inf_ptr = tolower(*inf_ptr);
+
+            if (0 == strcmp(inf_str, "-inf")) {
```

Comment:
also not sure if this will be more clear to make a if/else if for len 3 and another for len 4 (I 'c' lots of c code doing things like this)